internal/graph/storagewrapper.go

```diff
@@ -404,8 +508,9 @@ func (c *cachedIterator) flush() {
 	// Copy tuples buffer into new destination before storing into cache
 	// otherwise, the cache will be storing pointers. This should also help
 	// with garbage collection.
-	tuples := make([]*openfgav1.Tuple, len(c.tuples))
+	tuples := make([]*storage.TupleRecord, len(c.tuples))
```

Comment:
We could do:
```
tuples := c.tuples
c.tuples := nil
```

Thus the pointer gets de-referenced from the struct (so it can get gc'ed) but we can still store it without having to pay the penalty of the `copy` operation.